### PR TITLE
Data preprocessing

### PR DESCRIPTION
**Pull request by EdAguilarB**
## Description
Creation of UI of data preprocessing tab. Only interface with no functionality has been created.

## Linked issues
- #244 

## (Optional) Screenshots
image
image
image

### CODE REVIEW

**Review by dcl10 — CHANGES_REQUESTED** [review]

File: biofefi/pages/2_Data_Preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+def run_preprocessing(data: pd.DataFrame, experiment_path: Path) -> None:
+
+    X = data.iloc[:, :-1]
+    y = data.iloc[:, -1]
+
+    config = build_config()
+    save_options(data_preprocessing_options_path(experiment_path), config)
+    X = normalise_independent_variables(config.dependent_variable_normalisation, X)
+    y = transform_dependent_variable(config.independent_variable_transformation, y)
```
Comment (dcl10) [inline biofefi/pages/2_Data_Preprocessing.py:152]: should these be the other way around? `normalise_independent_variables(config.independent_variable_normalisation, X)` and `transform_dependent_variable(config.dependent_variable_transformation, y)`

**Review by dcl10 — CHANGES_REQUESTED** [review]

File: biofefi/options/file_paths.py [hunk 2]
```diff
@@ -14,6 +14,20 @@ def uploaded_file_path(file_name: str, experiment_path: Path) -> Path:
     return experiment_path / file_name
 
 
+def raw_data_path(file_name: str, experiment_path: Path) -> Path:
+    """Create the full path to the directory to save raw data.
+
+    Args:
```
Comment (dcl10) [inline biofefi/options/file_paths.py:20]: add `file_name` to the docstring please. Also create a test please. The other functions in this model have tests too. Check out `tests/options/test_file_paths.py`

File: biofefi/options/file_paths.py [hunk 3]
```diff
@@ -187,6 +201,26 @@ def execution_options_path(experiment_path: Path) -> Path:
     return experiment_path / "execution_options.json"
 
 
+def data_preprocessing_options_path(experiment_path: Path) -> Path:
```
Comment (dcl10) [inline biofefi/options/file_paths.py:204]: Like, `raw_data_path` can you add a test in `tests/options/test_file_paths.py` please

**Review by dcl10 — CHANGES_REQUESTED** [review]

File: biofefi/options/choices.py [hunk 4]
```diff
@@ -44,7 +44,7 @@
 TRANSFORMATIONSY = [
```
[the remark this answers is not in the document]
Comment (EdAguilarB) [reply biofefi/options/choices.py:44]: Oki! Would you like me to rename the `TransformationsY(StrEnum)` from `biofefi.options.enums`?

File: biofefi/options/file_paths.py
[the remark this answers is not in the document]
Comment (EdAguilarB) [reply biofefi/options/file_paths.py:1]: I did not understand this :(

File: biofefi/options/choices.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dcl10) [reply biofefi/options/choices.py:44]: No that one's fine. The PascalCase makes it clear that it's two words

File: biofefi/options/file_paths.py
[the remark this answers is not in the document]
Comment (dcl10) [reply biofefi/options/file_paths.py:1]: Okay, I'll try and have a call with you a little later today

File: biofefi/pages/2_Data_Preprocessing.py [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+            ConfigStateKeys.DependentNormalisation
+        ].lower(),
+        independent_variable_transformation=st.session_state[
+            ConfigStateKeys.IndependentNormalisation
+        ].lower(),
+    )
+    return preprocessing_options
+
+
+def run_feature_selection(feature_selection_methods: dict, data: pd.DataFrame) -> None:
```
Comment (dcl10) [inline biofefi/pages/2_Data_Preprocessing.py:57]: 1. I don't like the fact that this reassigning `data` within this function. I think you should return build a new dataframe and return that instead.
2. Docstring
3. Could this function go into a module under the services folder please?

File: biofefi/pages/2_Data_Preprocessing.py [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+    if feature_selection_methods[ConfigStateKeys.LassoFeatureSelection]:
+        lasso = Lasso(alpha=st.session_state[ConfigStateKeys.RegularisationTerm])
+        lasso.fit(X, y)
+        selected_features = X.columns[lasso.coef_ != 0]
+        X = X[selected_features]
+
+    data = pd.concat([X, y], axis=1)
+
+
+def normalise_independent_variables(normalisation_method: str, X):
```
Comment (dcl10) [inline biofefi/pages/2_Data_Preprocessing.py:94]: 1. Docstring
2. Move to a module under services please

File: biofefi/pages/2_Data_Preprocessing.py [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+
+    column_names = X.columns
+    X = scaler.fit_transform(X)
+
+    X = pd.DataFrame(X, columns=column_names)
+
+    return X
+
+
+def transform_dependent_variable(transformation_y_method: str, y):
```
Comment (dcl10) [inline biofefi/pages/2_Data_Preprocessing.py:113]: 1. Docstring
2. Move to a module under services please

File: biofefi/pages/2_Data_Preprocessing.py [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+    elif transformation_y_method == TransformationsY.StandardisationNormalisation:
+        scaler = StandardScaler()
+        y = scaler.fit_transform(y)
+
+    y = pd.DataFrame(y, columns=[column_name])
+
+    return y
+
+
+def run_preprocessing(data: pd.DataFrame, experiment_path: Path) -> None:
```
Comment (dcl10) [inline biofefi/pages/2_Data_Preprocessing.py:144]: 1. Docstring
2. Move to a module under services please

**Review by dcl10 — CHANGES_REQUESTED** [review]

**Review by dcl10 — APPROVED** [review]
